pkg/storage/bucket/bucket_util.go

```diff
@@ -7,27 +7,49 @@ import (
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/thanos-io/objstore"
+	"go.uber.org/atomic"
+
+	"github.com/cortexproject/cortex/pkg/util/concurrency"
 )
 
 // DeletePrefix removes all objects with given prefix, recursively.
 // It returns number of deleted objects.
 // If deletion of any object fails, it returns error and stops.
 func DeletePrefix(ctx context.Context, bkt objstore.Bucket, prefix string, logger log.Logger) (int, error) {
```

Comment:
Can we just remove it? It doesnt look like is being used anymore. I think if someone wants to avoid concurrency it can call with 1 as you are doing.